### PR TITLE
docs: Add a section on polymorphic queries.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Provide a description of polymorphic nested queries and a usage example
in the "Overview" section. Ensure that this is searchable and
discoverable by a user looking for polymorphic or nested queries.

### CODE REVIEW

File: docs/edgeql/overview.rst [hunk 1]
```diff
@@ -229,3 +229,74 @@ A query is evaluated recursively using the following procedure:
...
+
+
+.. _ref_eql_polymorphic_queries:
+
+Polymorphic Queries
+===================
+
+:index: nested shapes
+
+Sometimes it is useful to have an abstract type as a link target, thus
```
Comment (elprans) [inline docs/edgeql/overview.rst]: Should we also mention unions here?

Comment (vpetrovykh) [reply docs/edgeql/overview.rst]: We could _mention_ type unions and link to it without specifically blowing up this example. The type union section has its own polymorphic example already (the only one prior to this and not searchable).

Comment (1st1) [reply docs/edgeql/overview.rst]: I don't like words like "sometimes" in reference documentation.  I'd replace with "A link target can be an abstract type ... "

File: docs/edgeql/overview.rst [hunk 2]
```diff
@@ -229,3 +229,74 @@ A query is evaluated recursively using the following procedure:
...
+
+    type Game extending Named {
+        property price -> int64;
+    }
+
+    type Article extending Named {
+        property url -> str;
+    }
+
+The above schema describes some platform where there are users, games
```
Comment (1st1) [inline docs/edgeql/overview.rst]: "some platform" reads weird. I'd drop this sentence. The next one `Each User can add any of these entities` can be edited to "Every User`can have its `favorites` link point to either other User, Game, or Article"

File: docs/edgeql/overview.rst [hunk 3]
```diff
@@ -229,3 +229,74 @@ A query is evaluated recursively using the following procedure:
...
+and articles.  Each ``User`` can add any of these entities to
+``favorites``.  A natural question to ask then is how to fetch all this
```
Comment (1st1) [inline docs/edgeql/overview.rst]: "A natural question to ask then is how" -- "To fetch data related to different types of objects in the 'favorites' link the following syntax can be used"?

File: docs/edgeql/overview.rst [hunk 4]
```diff
@@ -229,3 +229,74 @@ A query is evaluated recursively using the following procedure:
...
+            # does my favorite User have?
+            favorites_count := count(
+                # start the path at the root of the shape
+                User.favorites[IS User].favorites)
+        }
+    }
+
+The :eql:op:`[IS TypeName] <ISINTERSECT>` construct can be used in
+:ref:`paths <ref_eql_expr_paths>` to restrict the target to a specific
+type.  When it is used in :ref:`shapes <ref_eql_expr_shapes>` it
```
Comment (1st1) [inline docs/edgeql/overview.rst]: "When it is used" sentence is redundant here. You literally just showed that 20 words earlier.

File: docs/edgeql/overview.rst (hunk 3 above)
Comment (vpetrovykh) [reply docs/edgeql/overview.rst]: OK

File: docs/edgeql/overview.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/edgeql/overview.rst]: OK

File: docs/edgeql/overview.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/edgeql/overview.rst]: OK

File: docs/edgeql/overview.rst (hunk 4 above)
Comment (vpetrovykh) [reply docs/edgeql/overview.rst]: This is deliberate. It is the first time the `[IS type]` is used or mentioned. Given that this is the "overview" section there's no reason to believe that the reader is already familiar with it. So the paragraph is meant to reinforce the idea that there are, in fact, two different usages of this syntax (both of which appear in the example).

File: docs/edgeql/overview.rst [hunk 5]
```diff
@@ -229,3 +229,91 @@ A query is evaluated recursively using the following procedure:
...
 
 6. Collect the results of all iterations to obtain the final result set.
+
+
+.. _ref_eql_polymorphic_queries:
+
+Polymorphic Queries
+===================
+
+:index: nested shapes
```
Comment (1st1) [inline docs/edgeql/overview.rst]: add "polymorphism"?

Comment (1st1) [reply docs/edgeql/overview.rst]: also "poly"

Comment (vpetrovykh) [reply docs/edgeql/overview.rst]: Done.

**Pull request merged**